app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBHelper.java

```diff
@@ -69,6 +73,34 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         db.endTransaction();
       }
     }
+    if (oldVersion < 5) {
+      // Add display order column and add arbitrary order to existing translations
+      db.beginTransaction();
+      try {
+        db.execSQL("ALTER TABLE "
+            + TranslationsTable.TABLE_NAME
+            + " ADD COLUMN "
+            + TranslationsTable.DISPLAY_ORDER
+            + " integer not null default -1"
+        );
+        Cursor translations = db.query(
```

Comment:
I am not entirely sure if `db.endTransaction()` auto-closes previous cursors, but you may need to close this cursor as well.

Reply:
+1 - also, as a hack, can we just use the row id as the value here so we don't have to do an upload loop?
i.e. something like

```sql
UPDATE translations SET display_order = id
```